core-bundle/src/Twig/Runtime/PictureConfigurationRuntime.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig\Runtime;
+
+use Contao\Image\PictureConfiguration;
+use Contao\Image\PictureConfigurationItem;
+use Contao\Image\ResizeConfiguration;
+use Twig\Extension\RuntimeExtensionInterface;
+
+final class PictureConfigurationRuntime implements RuntimeExtensionInterface
+{
+    /**
+     * Create a picture configuration from an array. This is intended to be
+     * used from within templates where programmatic building isn't available.
+     */
+    public function fromArray(array $configArray): PictureConfiguration
+    {
+        $config = new PictureConfiguration();
+
+        $config->setSize($this->createConfigItem($configArray));
+        $config->setFormats($configArray['formats'] ?? []);
+
+        $config->setSizeItems(
+            array_map(
+                function (array $imageSizeItem): PictureConfigurationItem {
+                    return $this->createConfigItem($imageSizeItem);
+                },
+                $configArray['items'] ?? []
+            )
+        );
+
```

Comment:
Should we throw an exception if `$configArray` has unused/invalid array keys? Would make it easier to spot typos, but would make it harder to write forwards-comaptible code (if new keys get added in the future).

Reply:
Yeah, I thought about this as well. It's probably a good idea to throw.

Perhaps we could use `PropertyAccess` as well here to set the values (after applying some minor transformations like `zoom` &rarr; `zoomLevel`). Then the code in the core would not need to know about possibly new config options in `PictureConfiguration`/`PictureConfigurationItem` and if you want to use the features you'd only have to add a dependency to a newer `contao/image` version. :thinking:

Reply:
I implemented said strategy in 89e114313402d8c5250bb69f981488655047cc51.